src/main/java/com/google/cloud/tools/maven/GenRepoInfoFileMojo.java

```
@@ -0,0 +1,85 @@
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+/**
+ * Generates repository information files for the Stackdriver Debugger.
+ */
+@Mojo(name = "genRepoInfoFile")
+@Execute(phase = LifecyclePhase.PREPARE_PACKAGE)
+public class GenRepoInfoFileMojo extends CloudSdkMojo implements GenRepoInfoFileConfiguration {
+
+  /**
+   * The root directory containing the source code of the app. Expected to be contained in a git
+   * repository.
+   */
+  @Parameter(alias = "genRepoInfoFile.sourceDirectory", defaultValue = "${project.basedir}",
+      property = "app.genRepoInfoFile.sourceDirectory")
+  protected File sourceDirectory;
+
+  /**
+   * Directory where the source context files will be generated.
+   */
+  @Parameter(alias = "genRepoInfoFile.outputDirectory",
+      defaultValue = "${project.build.outputDirectory}",
+      property = "app.genRepoInfoFile.outputDirectory")
+  protected String outputDirectory;
+
+  /**
+   * If {@code true}, ignores errors generating the source context files and proceeds to deployment.
+   * If {@code false}, the goal is aborted by generation errors.
+   */
+  @Parameter(alias = "genRepoInfoFile.ignoreErrors", defaultValue = "false",
+      property = "app.genRepoInfoFile.ignoreSrcCtxError")
+  protected boolean ignoreErrors;
+
+  @Override
+  public void execute() throws MojoExecutionException, MojoFailureException {
+    try {
+      getAppEngineFactory().genRepoInfoFile().generate(this);
+    } catch (AppEngineException aee) {
+      if (!ignoreErrors) {
+        throw new MojoExecutionException("An error occurred while generating source context files."
+            + " To ignore source context generation errors and proceed to deployment, use the "
```

In the unlikely event of there being an error, I think it would be better to let the user know about it more visibly, rather than moving on and possibly hiding the exception in the middle of all the other logging rubble, no?